nodedb.go

```diff
@@ -126,6 +126,10 @@ func (ndb *nodeDB) GetNode(hash []byte) *Node {
 
 // SaveNode saves a node to disk.
 func (ndb *nodeDB) SaveNode(node *Node, flushToDisk bool) {
+	ndb.saveNodeBatch(node, flushToDisk, ndb.recentBatch, ndb.snapshotBatch)
+}
+
+func (ndb *nodeDB) saveNodeBatch(node *Node, flushToDisk bool, rb, sb dbm.Batch) {
```

Comment:
Created an aux method so we can manually provide batch objects to not clobber the existing batch objects on the actual `nodeDB` when flushing a previous version.

Note to authors/maintainers: Having the batch objects be part of the actual `nodeDB` seems like a code smell and might warrant an architecture redesign.

Reply:
Good point, I'll add the comment as a godoc.